api/handlers.go

```diff
@@ -165,12 +228,67 @@ func (api *API) trackDownloadHandler(w http.ResponseWriter, req *http.Request, p
 		api.WriteError(w, err, http.StatusInternalServerError)
 		return
 	}
+
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	// If the size of the download is not supplied, assume the entire file was
+	// downloaded.
+	if downloadedBytes == 0 && skylink.Size > 0 {
```

Comment:
If `downloadedBytes == 0` then we return above so this should never be true

Reply:
Oh, good catch! I was sure I deleted this block of code but it has snuck back in somehow...

Originally, when I received no size for the download I assumed that it's a full download. Then it turned out browsers make some zero-sized requests, so I changed it.